isotope-intensity-estimation.Rmd

```diff
@@ -118,6 +113,153 @@ Park et al. 2008 but for checking if a candidate cluster is a isotope cluster,
+upper atom count for each element based on the mass (or m/z) of a tentative
+isotopologue peak (or for a compound with unknown chemical formula). These
+counts can then be used to calculate (lower and upper bounds of) isotope
+probabilities $P(I_{i})$ and subsequently a lower and upper bound for the
+intensity ratio $R_{I}$. This information hence allows to determine whether the
+tested peak's intensity matches the intensity expected from the isotopic
+substitution (within given boundaries).
+
+
+## Limitations
+
+- The present definitions base on compounds from HMDB and the presented approach
+  would hence fail to identify isotopes of e.g. anorganic compounds. Parameters
+  would have to be estimated based on corresponding collection of compounds.
+- By using only the most frequent isotopic substitutions we might miss detection
+  of less frequent isotopologues.
+
+## Parameter estimation
+
+### Identification of the most frequent isotopic substitutions {#isotopologues}
+
+### Definition of mass-dependent element counts {#elementcounts}
+
+# Evaluation
+
+
+
+## Questions and discussion
+
+- [ ] Change the way to describe isotopic substitutions: use [13]C2 instead of
+      2[13C] to describe 2 isotopes C13.
+- [ ] Use `isopattern(..., rel_to = 4)` which uses intensities relative to the
+      monoisotopic peak.
+- So far we subset the isotopic substitutions to the most frequently observed
+  ones. Should we maybe also consider the intensity of the resulting
+  isotopologue? Maybe a plot frequency against average intensity would be
+  helpful.
+- **Alternative approach**: for the evaluation whether a peak is from an
+  isotopologue we need: the mass difference $md$ and an intensity boundary. The
+  latter depends on the ratio of the probabilities of the monoisotopic peak to
+  the isotopologue $R_{I} = P(I)/P(0)$. With $P(I)/P(0) = x_{I}/x_{0}$ and
+  knowing the intensity of the monoisotopic peak $x_{0}$ we can get an estimate
+  of the isotopologues intensity $x_{I} = x_{0} * R_{I}$. From the
+  `enviPar::isopattern` function we should in fact already get everything we
+  need: the returned `matrix` contains the peaks' m/z and the abundances which
+  are equivalent to the probabilities. The $md$ for an isotopologue can be
+  calculated from the column `"m/z"` between the monoisotopic peak (first row)
+  and any of the isotopologues and the $R_{I}$ can be calculated from the column
+  `"abundance"` (ratio between any row and the first row). Thus we could:
+  - calculate the isopattern for each compound in HMDB. Then for each compound:
+  - calculate $md$ and $R_{I}$ for each isotopologue peak
+  - define the character string of the isotopic substitution
+    (e.g. `"[13]C1[16]O2"`).
+  - save this information along with the m/z of the monoisotopic peak to a
+    `data.frame` (or any other object?).
+  - Select the most frequent isotopic substitutions. For each of them: plot
```

Comment:
Yes, this approach makes sense. Since we compute the isotope pattern for each compound we can compute the ratios from them without using the functions I implemented before (at that time unfortunately I hadn't looked at enviPat yet). The only "problem" I see with the last point of the approach is that the relationship between R and mz for a given substitution won't be linear in general (and will vary from one to the other) and the modelling with linear regression might not be good. For example for the 3[C13] (2[C13]) substitution, R will be cubic (quadratic) in the counts of C and therefore I expect it to be cubic (quadratic) also in mz (since the relation between counts of C and mz is approximately linear). Anyway, I'll think better about it